src/main/java/Duke/Main/TaskList.java

```diff
@@ -0,0 +1,129 @@
+package duke.main;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import duke.errors.InvalidIndexException;
+import duke.tasks.Task;
+
+/**
+ * The type Task list.
+ */
+public class TaskList {
+    /**
+     * The Storage.
+     */
+    private Storage storage;
+
+    /**
+     * Instantiates a new Task list.
+     *
+     * @param storage the storage
+     */
+    public TaskList(Storage storage) {
+        this.storage = storage;
+        /*
+           - Values are constantly updated to prevent User's data
+           from disappearing if they quit halfway through.
+           - This however reduces the efficiency as it has to
+           pull and push to the txt file each time.
+         */
```

Comment:
I like this documentation- clearly explained the mechanism, advantages and disadvantages. 👍